File: src/main/java/io/github/asyncronous/apothecary/handler/PoisonedBladeHandler.java
package io.github.asyncronous.apothecary.handler;

import cpw.mods.fml.common.eventhandler.SubscribeEvent;
import net.minecraft.entity.EntityLiving;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.potion.PotionEffect;
import net.minecraftforge.event.entity.player.AttackEntityEvent;

public final class PoisonedBladeHandler{
    @SubscribeEvent
    public void onPlayerAttack(AttackEntityEvent e){
        ItemStack currIS = e.entityPlayer.getCurrentEquippedItem();
        if(currIS.hasTagCompound()){
            NBTTagCompound comp = currIS.stackTagCompound;

            if(comp.getBoolean("poisoned")){
                if(e.target instanceof EntityLiving){
                    EntityLiving l = (EntityLiving) e.target;
                    l.addPotionEffect(new PotionEffect(comp.getByte("poisonId"), 200));
                    comp.setBoolean("poisoned", false);
                }
            }
        }
    }
}

File: src/main/scala/io/github/asyncronous/apothecary/handler/PoisonedBladeHandler.scala
package io.github.asyncronous.apothecary.handler

import cpw.mods.fml.common.eventhandler.SubscribeEvent
import io.github.asyncronous.apothecary.ApothecaryTag
import net.minecraft.entity.EntityLiving
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.potion.PotionEffect
import net.minecraftforge.event.entity.player.AttackEntityEvent

object PoisonedBladeHandler{
  @SubscribeEvent
  def onPlayerAttack(e: AttackEntityEvent): Unit ={
    val stack: ItemStack = e.entityPlayer.getCurrentEquippedItem();
    if(stack.hasTagCompound() && stack.getTagCompound().hasKey(ApothecaryTag.IDENTIFIER)){
      val comp: NBTTagCompound = ApothecaryTag.getTag(stack);
      if(comp.getBoolean("poisoned")){
        if(e.target.isInstanceOf[EntityLiving]){
          val living: EntityLiving = e.target.asInstanceOf[EntityLiving];
          living.addPotionEffect(new PotionEffect(comp.getByte("poisonId"), 200));
          comp.setBoolean("poisoned", false);
        }
      }
    }
  }
}